### PR TITLE
chore: add Package Manager related notes

### PR DESCRIPTION
**Pull request by 0618**
## Problem

**Issue number, if available:**

## Changes

**Corresponding docs PR, if applicable:**

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: packages/create-amplify/README.md [hunk 1]
```diff
@@ -19,3 +19,19 @@ In a frontend project folder or empty folder, run `npm create amplify`.
    a. Do not install or use `amplify` globally
    b. Use the same package manager for all commands in a given project
    c. When running E2E tests locally without a package manager, set `npm_config_user_agent` to `npm`.
+
+# Manual Alternative
```
Comment (Amplifiyer) [inline packages/create-amplify/README.md:23]: This table is great for comparison of different package managers and finding their differences, maybe not so much as steps that customers can follow for their package manager.

I think this belongs in the docs site with a section per package manager to list commands/steps to run for scaffolding an amplify project. what do you think @josefaidt?

File: PROJECT_ARCHITECTURE.md [hunk 2]
```diff
@@ -199,3 +199,9 @@ Under the hood, the backend deployer executes a command like `npx cdk synth --ap
 29. The BackendFactory composes all of the constructs into the Backend object
 30. The BackendFactory returns the Backend object to the caller (the `amplify/backend.ts` file)
 31. This is the object that customers can then interact with to add overrides and custom resources to their project. From this point, the `backend.ts` file is essentially a "normal" CDK app. L1 and L2 resources can be accessed on the backend object and stacks can be created using `backend.createStack()`.
+
+# Package Manager Support
```
[the remark this answers is not in the document]
Comment (edwardfoyle) [reply PROJECT_ARCHITECTURE.md:203]: Agreed, although that probably belongs in the readme, not the project arch doc

[the remark this answers is not in the document]
Comment (0618) [reply PROJECT_ARCHITECTURE.md:203]: I was thinking of packages/create-amplify/README.md, but we also use them in pipeline-deploy and sandbox. Should we add it to the top README of the repo?

File: PROJECT_ARCHITECTURE.md [hunk 3]
```diff
@@ -199,3 +199,9 @@ Under the hood, the backend deployer executes a command like `npx cdk synth --ap
...
+
+- We use `npm_config_user_agent` to decide which package manager the user is using.
+- Yarn only supports NodeJS 20 because of the NodeJS [loader bug](https://github.com/nodejs/node/pull/43772) causing yarn [chain loader error](https://github.com/yarnpkg/berry/issues/4694).
```
[the remark this answers is not in the document]
Comment (0618) [reply PROJECT_ARCHITECTURE.md]: Done 59c84c7bb9

File: PROJECT_ARCHITECTURE.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (0618) [reply PROJECT_ARCHITECTURE.md:203]: Done 59c84c7bb9

File: packages/create-amplify/README.md (hunk 1 above)
Comment (josefaidt) [reply packages/create-amplify/README.md:23]: is this section necessary for the readme? we can have a "manual install" page in the docs but this is very granular and we only document npm flows in the docs site

Comment (0618) [reply packages/create-amplify/README.md:23]: In that case we can remove this section from the PR

Comment (0618) [reply packages/create-amplify/README.md]: Removed 21a56b2947

**Review by edwardfoyle — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -10,6 +10,11 @@ To quickly get started with Amplify (Gen 2), please visit [AWS Amplify (Gen 2) Q
 npm create amplify@latest
 ```
 
+## Package Manager Support
+
+- NodeJS 20 or later version is required for Yarn because of the NodeJS [loader bug](https://github.com/nodejs/node/pull/43772) causing yarn [chain loader error](https://github.com/yarnpkg/berry/issues/4694).
+- Yarn PnP is not supported. Run `yarn config set nodeLinker node-modules` to use "node-modules".
```
Comment (Amplifiyer) [inline README.md:16]: This still doesn't have a list of package managers that are supported.

Comment (0618) [reply README.md]: awhhhh....good catch! done 5c744d5c4e

**Review by Amplifiyer — APPROVED** [review]

**Review by edwardfoyle — APPROVED** [review]

**Pull request merged**